Enterprise-CMCS/managed-care-review
# Fix: Error Page display (post sidebar) and Headings

## Summary
- Fix design bugs from recent sprint work with error page displays and the page heading (the darker blue row in the header)

Other changes that enabled this:
- Deleted `PageHeading` component. It's not needed – we have another way we handle `h1` announcing now - which is via the setting focus on the Heading with `updateHeading`. That both updates the heading visually and resets the focus for a11y. 
- Add `contractName` to the contract revision data coming back from graphql. Did this to ensure this is always the same with contract name API rather than generating on the fly  (this leads to accessing state programs json all over the frontend code). 

#### Related issues
Design bugs came from recent work with #2296 and moving the sidebar to state submission form

#### Screenshots
**BEFORE**
![Screenshot 2024-03-06 at 3 34 54 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/2c7aff14-2a10-4916-806c-013c7b956e36)
![Screenshot 2024-03-06 at 3 47 31 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/2674b062-e391-42a3-bf86-cb91ffc174c2)

**AFTER**
![Screenshot 2024-03-06 at 3 35 48 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/76713588-7352-4571-a2bd-c5cf7bc704d9)
![Screenshot 2024-03-06 at 3 46 03 PM](https://github.com/Enterprise-CMCS/managed-care-review/assets/10750442/9f0a8be0-15f6-4d1e-b64d-668263b2894c)

#### Test cases covered
Did not add tests



## QA guidance
- Trigger error pages in the app (not found or fetching draft submission as CMS user or fetching forbidden submission as state use)
- **Check that v2 Rate Details. Header should look correct and have the same contract name as that submission when its in the dashboard**
